spec/query_objects/filtered_articles_query_spec.rb

```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+
+RSpec.describe FilteredArticlesQuery do
+  subject(:all) { query.all }
+
+  let(:query) { described_class.new(relation, filter_params) }
+  let(:relation) { Article.all }
+
+  context "with no params" do
+    let(:filter_params) { {} }
+
+    it { is_expected.to match_array(relation) }
+  end
+
+  describe "search param" do
+    let(:filter_params) { { search: "search" } }
+    let!(:target_articles) do
```

Comment:
why you use one `let` for set upping articles? seems like, you can use separate `let` to simplify build expected result. I mean
```ruby
let(:article_1) { create(:article, :published, title: "How to add search to Rails project?") }
let(:article_2) { create(:article, :published, content: "Why you not use pg_search?") }
let(:article_3) { create(:article, :published, content: "Invalid") }
let(:article_4) { create(:article, :not_published, content: "Search") } 

let(:expected_result) { [article_1, article_2] }

# ...
  it { is_expected.to match_array(expected_result) }
```